node/hyperbahn/service_proxy.js

```diff
@@ -296,12 +296,31 @@ function refreshServicePeer(serviceName, hostPort) {
 ServiceDispatchHandler.prototype.removeServicePeer =
 function removeServicePeer(serviceName, hostPort) {
     var self = this;
+
     var svcchan = self.channel.subChannels[serviceName];
-    var peer = svcchan.peers.get(hostPort);
-    if (peer) {
+    if (!svcchan) {
+        return;
+    }
+
+    var peer = self.channel.peers.get(hostPort);
+    if (!peer) {
+        return;
+    }
+
+    var anyOtherSubChan = false;
+    var subChanKeys = Object.keys(self.channel.subChannels);
+    for (var i = 0; i < subChanKeys; i++) {
+        var subChan = self.channel.subChannels[subChanKeys[i]];
+        if (subChan.peers.get(hostPort)) {
+            anyOtherSubChan = true;
+            break;
+        }
+    }
+
+    if (!anyOtherSubChan) {
```

Comment:
so if there is any other sub channel, we don't do anything, meaning traffic will still go through. my original intention is to stop forwarding once we de-advertise.